box_connector.py

```diff
@@ -0,0 +1,245 @@
+# --
+# File: box_connector.py
+#
+# Copyright (c) 2021 Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+# either express or implied. See the License for the specific language governing permissions
+# and limitations under the License.
+#
+
+import json
+
+# Phantom App imports
+import phantom.app as phantom
+import phantom.rules as ph_rules
+import requests
+from phantom.action_result import ActionResult
+from phantom.base_connector import BaseConnector
+
+from box import box
+
+
+class RetVal(tuple):
+    def __new__(cls, val1, val2=None):
+        return tuple.__new__(RetVal, (val1, val2))
+
+
+class BoxConnector(BaseConnector):
+
+    def __init__(self):
+
+        # Call the BaseConnectors init first
+        super(BoxConnector, self).__init__()
+
+    def _handle_test_connectivity(self, param):
+
+        # Add an action result object to self (BaseConnector) to represent the action for this param
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        # NOTE: test connectivity does _NOT_ take any parameters
+        # i.e. the param dictionary passed to this handler will be empty.
+        # Also typically it does not add any data into an action_result either.
+        # The status and progress messages are more important.
+        access_token = self.box._get_auth_token()
+        if access_token is None:
+            self.save_progress("Whoops, something went wrong in retrieving a token: " + str(access_token))
+            return action_result.set_status(phantom.APP_ERROR)
+        headers = {"Authorization": "Bearer " + str(access_token)}
+
+        result = self.box._make_rest_call(url="https://api.box.com/2.0/folders/0", headers=headers)
+        if result.get("type") != "error":
+            self.save_progress("Test Connectivity Passed")
+            return action_result.set_status(phantom.APP_SUCCESS)
+        else:
+            self.save_progress("Test Connectivity Failed")
+            return action_result.set_status(phantom.APP_ERROR)
+
+    def _handle_upload_file(self, param):
+        action_result = self.add_action_result(ActionResult(dict(param)))
```

Comment:
Should we add some docstring for these handlers detailing what it does?
This should help us maintain these handlers easier.

Reply:
Added docstring for handlers.